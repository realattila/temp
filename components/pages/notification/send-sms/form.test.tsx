/**
 * @jest-environment jsdom
 */

import { render, cleanup, fireEvent, waitFor, act, logDOM } from '@testing-library/react';

import { rest } from 'msw';
import { setupServer } from 'msw/node';

import SendSmsNotificationForm from './form';

const server = setupServer(
  rest.post('http://localhost:3000/api/notification/SendNotification', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ greeting: 'hello there' }));
  }),
);

describe('test notification send sms Form', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules(); // Most important - it clears the cache
    process.env = { ...OLD_ENV, API_BASE_URL_NOTIFICATION_SERVICE: 'http://localhost:3000/' }; // Make a copy
  });

  afterAll(() => {
    process.env = OLD_ENV; // Restore old environment
  });

  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('success send sms', async () => {
    const listProviders = [
      {
        value: 1,
        key: 1,
      },
      {
        value: 2,
        key: 2,
      },
    ];
    const renderForm = render(<SendSmsNotificationForm providersList={listProviders} />);

    const selectInput = renderForm.getByLabelText('Default select example');
    fireEvent.change(selectInput, { target: { value: '1' } });

    const textInput = renderForm.getByPlaceholderText('form.phone.placeholder');
    fireEvent.change(textInput, { traget: { value: 'salam bar Iran' } });

    const textAreaInput = renderForm.getByPlaceholderText('form.body.placeholder');
    fireEvent.change(textAreaInput, { target: { value: 'salam bar Mohandes' } });

    const phoneInput = renderForm.getByPlaceholderText('form.phone.placeholder');
    fireEvent.change(phoneInput, { target: { value: '09038276860' } });

    await act(async () => {
      const button = await renderForm.getByText('form.submit');
      fireEvent.click(button);
    });

    await waitFor(async () => {
      expect(renderForm.getByText('form.success')).toBeInTheDocument();
    });
  });

  it('faild send Sms', async () => {
    const listProviders = [
      {
        value: 1,
        key: 1,
      },
      {
        value: 2,
        key: 2,
      },
    ];
    const renderForm = render(<SendSmsNotificationForm providersList={listProviders} />);

    const selectInput = renderForm.getByLabelText('Default select example');
    fireEvent.change(selectInput, { target: { value: '1' } });

    const textInput = renderForm.getByPlaceholderText('form.phone.placeholder');
    fireEvent.change(textInput, { traget: { value: 'salam bar Iran' } });

    const textAreaInput = renderForm.getByPlaceholderText('form.body.placeholder');
    fireEvent.change(textAreaInput, { target: { value: 'salam bar Mohandes' } });

    const phoneInput = renderForm.getByPlaceholderText('form.phone.placeholder');
    fireEvent.change(phoneInput, { target: { value: '09038276860' } });

    server.use(
      rest.post('http://localhost:3000/api/notification/SendNotification', (req, res, ctx) => {
        return res(ctx.status(500), ctx.json({ greeting: 'hello there' }));
      }),
    );
    await act(async () => {
      const button = await renderForm.getByText('form.submit');
      fireEvent.click(button);
    });

    await waitFor(async () => {
      expect(renderForm.getByRole('alert')).toBeInTheDocument();
    });
  });
});

export default {};
