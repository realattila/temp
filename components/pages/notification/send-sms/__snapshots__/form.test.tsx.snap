// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test notification send sms Form test send sms 1`] = `
<div
  className="bg-white p-4"
>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="d-flex flex-column gap-2"
    >
      <div>
        <label
          className="form-label"
          htmlFor="selecte-input_providerName"
        >
          form.provider.label
        </label>
        <select
          aria-label="Default select example"
          className="form-select"
          disabled={false}
          id="selecte-input_providerName"
          onChange={[Function]}
          value="-1"
        >
          <option
            value="-1"
          >
            
          </option>
          <option
            value={1}
          >
            1
          </option>
          <option
            value={2}
          >
            2
          </option>
        </select>
        <div
          className=""
        />
      </div>
      <div
        className="form-input-group"
      >
        <label
          className="form-label"
          htmlFor="text-input_Recipient"
        >
          form.phone.label
        </label>
        <input
          className="form-control  false"
          disabled={false}
          id="text-input_Recipient"
          onChange={[Function]}
          placeholder="form.phone.placeholder"
          required={false}
          type="text"
          value=""
        />
        <div
          className=""
        />
      </div>
      <div>
        <label
          className="form-label"
          htmlFor="text-area-input_Body"
        >
          form.body.label
        </label>
        <textarea
          className="form-control"
          disabled={false}
          id="text-area-input_Body"
          onChange={[Function]}
          placeholder="form.body.placeholder"
          value=""
        />
        <div
          className=""
        />
      </div>
      <button
        className="btn btn-primary btn-loading mt-4"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <div
          className="spinner-border text-undefined spinner-border-sm btn-loading-spinner"
          role="status"
        >
          <span
            className="visually-hidden"
          >
            Loading...
          </span>
        </div>
        form.submit
      </button>
    </div>
  </form>
</div>
`;
